src/evict/evict_page.c

```diff
@@ -459,14 +462,22 @@ __evict_review(
 		flags = *flagsp;
 
 		/*
-		 * Check for an append-only workload needing an in-memory
-		 * split; we can't do this earlier because in-memory splits
-		 * require exclusive access. If an in-memory split completes,
-		 * the page stays in memory and the tree is left in the desired
-		 * state: avoid the usual cleanup.
+		 * Check for an append-only workload needing an in-memory split;
+		 * we can't do this earlier because in-memory splits require
+		 * exclusive access. If an in-memory split completes, the page
+		 * stays in memory and the tree is left in the desired state:
+		 * avoid the usual cleanup.
 		 */
 		if (LF_ISSET(WT_EVICT_INMEM_SPLIT))
 			return (__wt_split_insert(session, ref));
+
+		/*
+		 * If we can't split the page, check if eviction is possible
+		 * for this file (eviction is turned off during some parts of
+		 * checkpoint).
+		 */
+		if (F_ISSET(btree, WT_BTREE_NO_EVICTION))
```

Comment:
@keithbostic The problem with this change is that the current code expects the ``WT_BTREE_NO_EVICTION`` flag to be ignored here when closing a file - @sulabhM is that correct?